packages/dev/core/src/Shaders/ShadersInclude/pbrDirectLightingSetupFunctions.fx

```diff
@@ -77,4 +85,145 @@ preLightingInfo computeHemisphericPreLightingInfo(vec4 lightData, vec3 V, vec3 N
     #endif
 
     return result;
-}
\ No newline at end of file
+}
+
+#ifdef AREALIGHTUSED

```

Comment:
That's practically the exact same code than in `lightsFragmentFunctions.fx`: shouldn't we factor it?